OPHD/States/MapViewStateIO.cpp

```diff
@@ -194,7 +194,8 @@ void MapViewState::load(const std::string& filePath)
 	readPopulation(root->firstChildElement("population"));
 	readTurns(root->firstChildElement("turns"));
 
-	XmlElement* energy = root->firstChildElement("energy");
+	auto energy = root->firstChildElement("energy");
+	if (!energy) { throw std::runtime_error("MapViewState::load(): Savegame file is missing '<energy>' tag."); }
```

Comment:
Similarly, a flatter structure seems possible here.

Reply:
Probably. Currently save games look like this:

```xml
<turns count="58" />
<energy energy="100" />
```

Which seems very inefficient to me. This could be done with a single element and a few attributes attached.